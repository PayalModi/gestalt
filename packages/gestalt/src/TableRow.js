// @flow strict
import * as React from 'react';
import styles from './Table.css';

type Props = {|
  children: React.Node,
  onClick?: ({
    event:
      | SyntheticMouseEvent<HTMLDivElement>
      | SyntheticKeyboardEvent<HTMLDivElement>,
  }) => void,
|};

const SPACE_CHAR_CODE = 32;
const ENTER_CHAR_CODE = 13;

export default function TableRow(props: Props) {
  const { children, onClick } = props;

  const handleKeyPress = (event: SyntheticKeyboardEvent<HTMLDivElement>) => {
    if (
      onClick &&
      (event.charCode === SPACE_CHAR_CODE || event.charCode === ENTER_CHAR_CODE)
    ) {
      // Prevent the default action to stop scrolling when space is pressed
      event.preventDefault();
      onClick({ event });
    }
  };

  const handleOnClick = (event: SyntheticMouseEvent<HTMLDivElement>) => {
    if (onClick) {
      onClick({ event });
    }
  };

  if (onClick) {
    return (
      <tr
        className={styles.trClickable}
        onClick={handleOnClick}
        onKeyPress={handleKeyPress}
        tabIndex="0"
        role="button"
      >
        {children}
      </tr>
    );
  }

  return <tr>{children}</tr>;
}
