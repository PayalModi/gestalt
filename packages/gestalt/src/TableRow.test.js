// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import TableRow from './TableRow.js';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <TableRow>
        <td>cell 1</td>
        <td>cell 2</td>
        <td>cell 3</td>
      </TableRow>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly when clickable', () => {
  const tree = renderer
    .create(
      <TableRow onClick={() => {}}>
        <td>cell 1</td>
        <td>cell 2</td>
        <td>cell 3</td>
      </TableRow>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
