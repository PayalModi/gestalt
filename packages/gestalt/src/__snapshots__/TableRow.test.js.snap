// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr>
  <td>
    cell 1
  </td>
  <td>
    cell 2
  </td>
  <td>
    cell 3
  </td>
</tr>
`;

exports[`renders correctly when clickable 1`] = `
<tr
  className="trClickable"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  <td>
    cell 1
  </td>
  <td>
    cell 2
  </td>
  <td>
    cell 3
  </td>
</tr>
`;
